packages/smooth_app/lib/pages/product/edit_ingredients_page.dart

```diff
@@ -0,0 +1,339 @@
+                        height: 400.0,
+                        color: Colors.black,
+                        child: Theme(
+                          data: darkTheme,
+                          child: DefaultTextStyle(
+                            style: const TextStyle(color: Colors.white),
+                            child: Padding(
+                              padding: const EdgeInsets.all(16.0),
+                              child: Column(
+                                children: <Widget>[
+                                  TextField(
+                                    enabled: !_updatingIngredients,
+                                    controller: _controller,
+                                    decoration: InputDecoration(
+                                      border: OutlineInputBorder(
+                                        borderRadius:
+                                            BorderRadius.circular(3.0),
+                                      ),
+                                    ),
+                                    maxLines: null,
+                                    textInputAction: TextInputAction.done,
+                                    onSubmitted: _onSubmitField,
+                                  ),
+                                  Text(
+                                      appLocalizations.ingredients_editing_instructions),
+                                ],
+                              ),
+                            ),
+                          ),
+                        ),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+        ],
+      ),
+    );
+  }
+}
+
+// The actions for the page in a row of FloatingActionButtons.
+class _ActionButtons extends StatelessWidget {
+  const _ActionButtons({
+    Key? key,
+    required this.hasImage,
+    required this.getImage,
+  }) : super(key: key);
+
+  final bool hasImage;
+  final VoidCallback getImage;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: <Widget>[
+        if (!hasImage)
+          FloatingActionButton.small(
+            tooltip: 'Take photo',
+            backgroundColor: Colors.white,
+            foregroundColor: Colors.grey,
+            onPressed: getImage,
+            child: const Icon(Icons.camera_alt),
+          ),
+        if (hasImage)
+          FloatingActionButton.small(
+            tooltip: 'Retake photo',
+            backgroundColor: Colors.white,
+            foregroundColor: Colors.grey,
+            onPressed: getImage,
+            child: const Icon(Icons.refresh),
+          ),
+        if (hasImage) const SizedBox(width: 12.0),
+        if (hasImage)
+          FloatingActionButton.small(
+            tooltip: 'Confirm',
+            backgroundColor: Colors.blue,
+            foregroundColor: Colors.white,
+            onPressed: () {},
```

Comment:
Nothing happens here?

Reply:
Whoops, should return to the previous page.